vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethod.java

```diff
@@ -79,6 +85,11 @@ public Void call() throws Exception {
 
 				log.debug("Invoking {}", reflectionMethod);
 				Object instance = event.getControllerInstance();
+				
+				Class<?> declaringClass = reflectionMethod.getDeclaringClass();
+				if (!declaringClass.isAssignableFrom(instance.getClass())) {
+					instance = container.instanceFor(declaringClass);
```

Comment:
I'm not getting why this is happening. When we forward the request a new flow of events should start, right? Why is  ExecuteMethod holding an instance to the previous controller? I think that if we use this solution other problems could still appear (anyone observing this event and making use of that controller instance could be bugged).